tests/python_client/testcases/test_index.py

```diff
@@ -377,455 +384,426 @@ def test_index_building_after_restart(self):
         """
         target: test create index interface
         method: create collection and add entities in it, create index
         expected: return search success
         """
-        connect.insert(collection, default_entities)
-        connect.create_index(collection, field_name, get_simple_index)
+        c_name = cf.gen_unique_str(prefix)
+        collection_w = self.init_collection_wrap(name=c_name)
+        data = cf.gen_default_list_data()
+        collection_w.insert(data=data)
         if get_simple_index["index_type"] != "FLAT":
-            index = connect.describe_index(collection, "")
-            create_target_index(get_simple_index, field_name)
-            assert index == get_simple_index
-
-    @pytest.mark.tags(CaseLabel.L0)
-    @pytest.mark.skip(reason="Repeat with test_index_field_name_not_existed")
-    def test_create_index_on_field_not_existed(self, connect, collection, get_simple_index):
-        """
-        target: test create index interface
-        method: create collection and add entities in it, create index on field not existed
-        expected: error raised
-        """
-        tmp_field_name = gen_unique_str()
-        result = connect.insert(collection, default_entities)
-        with pytest.raises(MilvusException) as e:
-            connect.create_index(collection, tmp_field_name, get_simple_index)
-
-    @pytest.mark.tags(CaseLabel.L2)
-    def test_create_index_on_field(self, connect, collection, get_simple_index):
+            collection_w.create_index(ct.default_float_vec_field_name, get_simple_index,
+                                  index_name=ct.default_index_name)
+            assert len(collection_w.indexes) == 1
+        
+        
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_create_index_non_existed_field(self):
         """
         target: test create index interface
         method: create collection and add entities in it, create index on other field
         expected: error raised
         """
-        tmp_field_name = "int64"
-        result = connect.insert(collection, default_entities)
-        connect.create_index(collection, tmp_field_name, get_simple_index)
+        c_name = cf.gen_unique_str(prefix)
+        collection_w = self.init_collection_wrap(name=c_name)
+        data = cf.gen_default_list_data()
+        collection_w.insert(data=data)
+        collection_w.create_index(ct.default_int8_field_name, default_index_params,
+                                  index_name=ct.default_index_name,
+                                  check_task=CheckTasks.err_res,
+                                  check_items={ct.err_code: 1, ct.err_msg: "cannot create index on non-existed field: int8"}
+                                  )
 
-    @pytest.mark.tags(CaseLabel.L2)
-    # @pytest.mark.timeout(BUILD_TIMEOUT)
-    def test_create_index_no_vectors(self, connect, collection, get_simple_index):
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_create_index_no_vectors(self):
```

Comment:
no vectors? It seems the code is not no vectors?

Reply:
> no vectors? It seems the code is not no vectors?

I think the index was created without vectors